### PR TITLE
#162725266 List users functionality

### PR DESCRIPTION
**Pull request by armstrongsouljah**
#### What does this PR do?
- Lists profiles of the existing authors. 
- Limits access to user profile list to only authenticated users. 

#### Description of the task to be implemented.
 - A logged in/authenticated user should be able to see the the list and profiles of existing authors. He 
  should however not be able to edit the profiles

#### How should this be manually tested?
- Fetch this branch to your local machine by running ` git fetch origin`
- Activate your virtual environment by running `source venv/bin/activate`
- Run your local server using `python manage.py runserver`
- Run the postman collections using the link below.

[Run in Postman](https://app.getpostman.com/run-collection/de6aea922b105498d670)

#### What are the relevant pivotal tracker stories?
[#162725266](https://www.pivotaltracker.com/story/show/162725266)

### CODE REVIEW

**Comment by MuhweziDeo** [comment]
which endpoint should I use to test the feature

File: authors/apps/profiles/tests.py [hunk 1]
```diff
@@ -27,3 +27,12 @@ def test_edit_my_profile_succeeds(self):
             HTTP_AUTHORIZATION='Bearer ' + self.test_user_token
             )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_authenticated_user_view_author_profiles_succeeds(self):
```
Comment (MuhweziDeo) [inline authors/apps/profiles/tests.py:54]: comments/documentation NEEDED here @armstrongsouljah

Comment (armstrongsouljah) [reply authors/apps/profiles/tests.py:54]: Thanks for spotting that out. Let me resolve that.

**Review by MuhweziDeo — DISMISSED** [review]
Add documentation and  all is good @armstrongsouljah

File: authors/apps/profiles/tests.py (hunk 1 above)
Comment (armstrongsouljah) [reply authors/apps/profiles/tests.py:54]: @MuhweziDeo I added the recommended docs in the test files.

**Review by MuhweziDeo — DISMISSED** [review]
@armstrongsouljah  All is now good

**Review by kdaglas — DISMISSED** [review]
finished testing this and all looks good...

File: authors/apps/profiles/serializers.py [hunk 2]
```diff
@@ -23,12 +23,51 @@ def get_username(self, obj):
...
             'image',
             'date_of_birth'
         ]
+   
+    
+    def get_username(self, obj):
+        return f"{obj.user.username}"
+
+    def to_representation(self, instance):
+        print("instance", instance)
```
Comment (ja-odur) [inline authors/apps/profiles/serializers.py]: @armstrongsouljah remove print statement and document function

Comment (armstrongsouljah) [reply authors/apps/profiles/serializers.py]: Thanks for spotting that. Let me resolve it right away.

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by malfahad — APPROVED** [review]

**Review by MuhweziDeo — APPROVED** [review]

**Review by d-rita — APPROVED** [review]
Seems fine

**Pull request merged**